### PR TITLE
Adding config option to enable/disable role detection

### PR DESCRIPTION
**Pull request by fabiocav**
Adding a host config setting to disable the automatic role/primary host detection and having the feature disabled in the end-to-end tests.

### CODE REVIEW

File: src/WebJobs.Script/Host/ScriptHost.cs [hunk 1]
```diff
@@ -245,11 +245,13 @@ protected virtual void Initialize()
                     // Disable core storage 
                     ScriptConfig.HostConfig.StorageConnectionString = null;
                 }
-
-                // Create the lease manager that will keep handle the primary host blob lease acquisition and renewal 
-                // and subscribe for change notifications.
-                _blobLeaseManager = BlobLeaseManager.Create(storageString, TimeSpan.FromSeconds(15), ScriptConfig.HostConfig.HostId, TraceWriter);
-                _blobLeaseManager.HasLeaseChanged += BlobLeaseManagerHasLeaseChanged;
+                else if (ScriptConfig.RoleDetectionEnabled)
+                {
```
Comment (fabiocav) [inline src/WebJobs.Script/Host/ScriptHost.cs]: This change also ensures we're only creating a `BlobLeaseManager` if a connection string is defined.

Comment (brettsam) [inline src/WebJobs.Script/Host/ScriptHost.cs]: If you've set RoleDetectionEnabled to true but not provided a ConnectionString, should we throw there instead? It feels like you've asked us to do something that we can't do and maybe a descriptive error would help someone diagnose the problem.

---
In reply to: [72854143](https://github.com/Azure/azure-webjobs-sdk-script/pull/536#discussion_r72854143) [](ancestors = 72854143)

**Comment by brettsam** [comment]
Just one (minor) comment. Otherwise looks good.

File: src/WebJobs.Script/Host/ScriptHost.cs (hunk 1 above)
Comment (fabiocav) [inline src/WebJobs.Script/Host/ScriptHost.cs]: Yeah, I debated adding that. I think it would make sense.

**Pull request merged**